src/activities/associateGrade/AssociateGradeEntity.js

```diff
@@ -68,11 +69,34 @@ export class AssociateGradeEntity extends Entity {
 		return typeof newGradeEntity !== 'undefined' && newGradeEntity.hasActionByName(Actions.activities.associateGrade.chooseType);
 	}
 
+	canGetCategories() {
+		const newGradeEntity = this._getNewGradeEntity();
+		return typeof newGradeEntity !== 'undefined' && newGradeEntity.hasActionByName(Actions.activities.associateGrade.getCategories);
```

Comment:
any reason we don't just do `return newGradeEntity && newGradeEntity.hasActionByName(Actions.activities.associateGrade.getCategories);`

Reply:
yeah, it's so that `canGetCategories` always returns true or false, not undefined. This is necessary for tests, which reminds me I need to add tests for these new functions.